modules/search/search-service/main.bicep

```diff
@@ -0,0 +1,274 @@
+param diagnosticSettingsName string = ''
+
+@description('Optional. Resource ID of the diagnostic storage account. For security reasons, it is recommended to set diagnostic settings to send data to either storage account, log analytics workspace or event hub.')
+param diagnosticStorageAccountId string = ''
+
+@description('Optional. Resource ID of the diagnostic log analytics workspace. For security reasons, it is recommended to set diagnostic settings to send data to either storage account, log analytics workspace or event hub.')
+param diagnosticWorkspaceId string = ''
+
+@description('Optional. Resource ID of the diagnostic event hub authorization rule for the Event Hubs namespace in which the event hub should be created or streamed to.')
+param diagnosticEventHubAuthorizationRuleId string = ''
+
+@description('Optional. Name of the diagnostic event hub within the namespace to which logs are streamed. Without this, an event hub is created for each log category. For security reasons, it is recommended to set diagnostic settings to send data to either storage account, log analytics workspace or event hub.')
+param diagnosticEventHubName string = ''
+
+@description('Optional. When set to true, calls to the search service will not be permitted to utilize API keys for authentication. This cannot be set to true if \'authOptions\' are defined.')
+param disableLocalAuth bool = true
+
+@description('Optional. Enable telemetry via the Customer Usage Attribution ID (GUID).')
+param enableDefaultTelemetry bool = true
+
+@description('Optional. Describes a policy that determines how resources within the search service are to be encrypted with Customer Managed Keys.')
+@allowed([
+  'Disabled'
+  'Enabled'
+  'Unspecified'
+])
+param cmkEnforcement string = 'Unspecified'
+
+@description('Optional. Applicable only for the standard3 SKU. You can set this property to enable up to 3 high density partitions that allow up to 1000 indexes, which is much higher than the maximum indexes allowed for any other SKU. For the standard3 SKU, the value is either \'default\' or \'highDensity\'. For all other SKUs, this value must be \'default\'.')
+@allowed([
+  'default'
+  'highDensity'
+])
+param hostingMode string = 'default'
+
+@description('Optional. Location for all Resources.')
+param location string = resourceGroup().location
+
+@description('Optional. Specify the type of lock.')
+@allowed([
+  ''
+  'CanNotDelete'
+  'ReadOnly'
+])
+param lock string = ''
+
+@description('Optional. Network specific rules that determine how the Azure Cognitive Search service may be reached.')
+param networkRuleSet object = {}
+
+@description('Optional. The number of partitions in the search service; if specified, it can be 1, 2, 3, 4, 6, or 12. Values greater than 1 are only valid for standard SKUs. For \'standard3\' services with hostingMode set to \'highDensity\', the allowed values are between 1 and 3.')
+@minValue(1)
+@maxValue(12)
+param partitionCount int = 1
```

Comment:
Can we test with higher partition count? Maybe in the common file

Reply:
Added to common test case as suggested, test was successful.